Build failed on OS X 10.11: SOL_SOCKET not defined

UWSGI is failing to build on OS X 10.11 with Xcode 7.0.1 for me. Built from uwsgi-2.0.11.1.tar.gz.

```
bash-3.2$ python uwsgiconfig.py --build --verbose
...
[thread 3] clang -c -O2 -I. -Wall -Werror -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fno-strict-aliasing -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-format -Wno-format-security -DUWSGI_HAS_IFADDRS -DUWSGI_ZLIB -mmacosx-version-min=10.5 -DUWSGI_LOCK_USE_OSX_SPINLOCK -DUWSGI_EVENT_USE_KQUEUE -DUWSGI_EVENT_TIMER_USE_KQUEUE -DUWSGI_EVENT_FILEMONITOR_USE_KQUEUE -I/usr/local/Cellar/pcre/8.37/include -DUWSGI_PCRE -DUWSGI_ROUTING -DUWSGI_UUID -DUWSGI_VERSION="\"2.0.11.1\"" -DUWSGI_VERSION_BASE="2" -DUWSGI_VERSION_MAJOR="0" -DUWSGI_VERSION_MINOR="11" -DUWSGI_VERSION_REVISION="1" -DUWSGI_VERSION_CUSTOM="\"\"" -DUWSGI_YAML -DUWSGI_JSON -DUWSGI_SSL -I/Applications/Xcode-7.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2 -DUWSGI_XML -DUWSGI_XML_LIBXML2 -DUWSGI_PLUGIN_DIR="\".\"" -DUWSGI_DECLARE_EMBEDDED_PLUGINS="UDEP(python);UDEP(gevent);[...snip...] -DUWSGI_BUILD_DATE="\"02 October 2015 18:31:50\"" -o core/socket.o core/socket.c
...
core/socket.c:713:28: error: use of undeclared identifier 'SOL_TCP'
                if (setsockopt(serverfd, SOL_TCP, TCP_FASTOPEN, (const void *) &uw...
```

Here's my build log: https://gist.github.com/apjanke/33f2c1d1552ce6c74aba

Looks sort of like #966. But OS X's `<sys/socket.h>` does define SOL_SOCKET, as far as I can tell. (And from looking at the file, it's not buried in a weird `#ifdef`.) Don't know why it's not picking it up.

```
bash-3.2$ grep SOL_SOCKET /usr/include/sys/socket.h
#define	SOL_SOCKET	0xffff		/* options for socket level */
```

Found this while testing the Homebrew uwsgi formula for 10.11 (homebrew/homebrew#44018), but same error happens if I just download and build UWSGI outside of Homebrew.

--- a/core/socket.c
+++ b/core/socket.c
@@ -710,6 +710,11 @@ int bind_to_tcp(char *socket_name, int listen_queue, char *tcp_port) {
 
 	if (uwsgi.tcp_fast_open) {
 #ifdef TCP_FASTOPEN
+
+    #ifndef SOL_TCP
+    #define SOL_TCP IPPROTO_TCP
+    #endif
+
 		if (setsockopt(serverfd, SOL_TCP, TCP_FASTOPEN, (const void *) &uwsgi.tcp_fast_open, sizeof(int)) < 0) {
 			uwsgi_error("TCP_FASTOPEN setsockopt()");
 		}

